### PR TITLE
Removed infinite scroll, removed JS navigation, added pagination

### PR DESCRIPTION
**Pull request by Martin-Alexander**
### Changes:

- The index page now uses pagination instead of infinite scroll. The design of the pagination UI is the same as what we were using for the non-JS version of the website.

- Before this change, navigating to other categories would use JS to replace the main content of the page without reloading the tab. Now, everything is its own page and navigation is done via normal links.

- Years are now technically categories. So, every time you make your first post of the year you have to also add a new category for that year.

- Since years are also categories, for every new post you make you must include the year (as a string) as a category.

- I have gone through all your past posts and added the year as a category.

### Concerns:

- Since the category pages don't use infinite scroll anymore the browser will try to load all images, videos, and widgets right off the bat rather than as the user scrolls down the page. Let me know what you think about the performance of this, and if you want it improved I can implement a "lazy loading" system whereby images (and maybe some videos) are only when they're about to come in view.

### CODE REVIEW

File: _includes/header.html [hunk 1]
```diff
@@ -1,3 +1,5 @@
+{% comment %} {% assign this_is_a_year_categoy_page = page.title contains "20" and page.title | size == 4%} {% endcomment %}
```
Comment (mattbischoff) [inline _includes/header.html:1]: Rather than a check like this, could we use the layout property on the year categories and set that to year?

Comment (Martin-Alexander) [reply _includes/header.html:1]: I'm not sure what you mean by "layout property".

But I was thinking of adding something like `year: true` in the front matter of each category in `_categories/`.

Did you have something like that in mind?

Comment (mattbischoff) [reply _includes/header.html:1]: I was talking about https://jekyllrb.com/docs/step-by-step/04-layouts/

Comment (Martin-Alexander) [reply _includes/header.html:1]: Sorry, I don't understand what you're proposing. 

The variable `site.categories` returns a list of categories with this structure:

```
[
  ["managment", <#Post>, <#Post>, ...],
  ["2020", <#Post>, <#Post>, ...],
  ["shorter", <#Post>, <#Post>, ...],
  ...
]
```

From this, I'm not sure where I could leave & then access any info on whether or not the category is year. I don't think categories themselves have any internal properties. They're just names as far as I can tell.

Comment (mattbischoff) [reply _includes/header.html:1]: Ah yeah, you're right. I thought they could.

**Review by mattbischoff — APPROVED** [review]

**Pull request merged**